pkg/fanal/analyzer/language/rust/cargo/cargo.go

```diff
@@ -106,7 +106,7 @@ func (a cargoAnalyzer) removeDevDependencies(fsys fs.FS, dir string, app *types.
 	cargoTOMLPath := filepath.Join(dir, types.CargoToml)
 	directDeps, err := a.parseCargoTOML(fsys, cargoTOMLPath)
 	if errors.Is(err, fs.ErrNotExist) {
-		log.Logger.Debugf("Cargo: %s not found", cargoTOMLPath)
+		// Already logged this error in tomlDependencies
 		return nil
```

Comment:
Yes, exactly...

Reply:
Hm... tests are failing with out this hack. Trying to find a better solution...

Reply:
I have question: do we need this check at all?

https://github.com/anfedotoff/trivy/blob/a0819b644715abed1505f0a3b7558d41f80b80db/pkg/fanal/analyzer/language/rust/cargo/cargo.go#L168-L170

I think, we will log this here:
https://github.com/anfedotoff/trivy/blob/a0819b644715abed1505f0a3b7558d41f80b80db/pkg/fanal/analyzer/language/rust/cargo/cargo.go#L71-L73

Reply:
If one of members doesn't exist, we continue to check the remaining members.
Without this check you will stop parsing remaining toml files.

Correct me, if i am wrong.

Reply:
> If one of members doesn't exist, we continue to check the remaining members. Without this check you will stop parsing remaining toml files.
> 
> Correct me, if i am wrong.

Yes, you are right. But I think we should do the following:

1.  Stop if it is not possible to parse the root (workspace) Cargo.toml
2. Continue if it is not possible to parse member Cargo.toml

I pushed some changes. Warnings in logs LGTM (I malformed some member field):

```
2023-11-09T15:03:03.345+0300	DEBUG	Walk the file tree rooted at '/home/fedotoff/casr' in parallel
2023-11-09T15:03:03.415+0300	WARN	Unable to parse "ibcasr/Cargo.toml": file open error: open ibcasr/Cargo.toml: file does not exist
```

P.S. Sorry, for being so slow. I'm changing work, trying to find some time for open source:).

Reply:
> Stop if it is not possible to parse the root (workspace) Cargo.toml

You can't continue parsing because in this case you don't have any members to parse 😄

But you are right. I like this logic.

Reply:
But I'm not sure if we should show warning if the root `Cargo.toml` file doesn't exist.
This can confuse users if they only scan the `cargo.lock` file.

I think we need to revert previous logic for root `cargo.toml` file (with debug message).